epochStart/metachain/legacySystemSCs.go

```diff
@@ -321,7 +323,12 @@ func (s *legacySystemSCProcessor) unStakeNodesWithNotEnoughFunds(
 			continue
 		}
 
-		validatorInfo.SetList(string(common.LeavingList))
+		validatorLeaving := validatorInfo.ShallowClone()
+		validatorLeaving.SetList(string(common.LeavingList))
```

Comment:
Yes, but this is because we had to force `ValidatorInfo` DTO into an interface. This `ValidatorInfo` uses proto which auto-generates structs with `List` field of type `string` => `GetList()` will return by default `string` upon proto file generation.

If I change `SetList`, I would also have to change `GetList`, which is an auto-generated func by proto(returns `string`)